packages/starknet-snap/src/state/account-state-manager.ts

```diff
@@ -435,6 +457,18 @@ export class AccountStateManager extends StateManager<AccContract> {
     );
   }
 
+  async isAccountNameExist(
+    { accountName, chainId }: { accountName: string; chainId: string },
+    state,
+  ): Promise<boolean> {
+    const snapState = state ?? (await this.get());
+    return (
+      this.getCollection(snapState).find(
+        (data) => data.accountName === accountName && data.chainId === chainId,
+      ) !== undefined
```

Comment:
same comment as above

```suggestion
      this.getCollection(snapState).find(
        (data) => data.accountName?.trim() === accountName && data.chainId === chainId,
      ) !== undefined
```

Reply:
btw does it has a handing on UI?